src/codec/zstd/decoder.rs

```diff
@@ -16,6 +16,16 @@ impl ZstdDecoder {
         }
     }
 
+    pub(crate) fn new_with_params(params: &[crate::zstd::DParameter]) -> Self {
```

Comment:
I think we should propagate the error from zstd?

Reply:
I'm not necessarily opposed to this, though in the interest of consistency, it appears that this library chooses to panic on construction errors for other encoders are decoders:

- https://github.com/Nullus157/async-compression/blob/7c2d5304b418254e13e810329ee0004a6484a7bd/src/codec/zstd/encoder.rs#L18
- https://github.com/Nullus157/async-compression/blob/7c2d5304b418254e13e810329ee0004a6484a7bd/src/codec/xz2/encoder.rs#L23

Reply:
That's unfortunate, maybe we should add a `try_` variant for these APIs?

cc @robjtede what's your thoughts on this?

Reply:
I think typically panics are acceptable when it comes to "programmer error" and I really, really don't expect decoding parameters to be user-input, or anything other than hardcoded. Therefore it's fine to me that this can panic. It's a fairly unrecoverable error for **both** these unwraps.